flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/InputChannelInfo.java

```diff
@@ -20,14 +20,15 @@
 import org.apache.flink.annotation.Internal;
 
 import java.io.Serializable;
+import java.util.Objects;
 
 /**
  * Identifies {@link org.apache.flink.runtime.io.network.partition.consumer.InputChannel} in a given subtask.
  * Note that {@link org.apache.flink.runtime.io.network.partition.consumer.InputChannelID InputChannelID}
  * can not be used because it is generated randomly.
  */
 @Internal
-public class InputChannelInfo implements Serializable {
+public final class InputChannelInfo implements Serializable {
```

Comment:
nitty nit: how does making those classes `final` make them easier to test? Maybe add a description in the commit message?